org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/ParentProcessWatcher.java

```diff
@@ -40,6 +42,9 @@ public final class ParentProcessWatcher implements Runnable, Function<MessageCon
 
 	public ParentProcessWatcher(LanguageServer server ) {
 		this.server = server;
+		if (ProcessHandle.current().parent().isPresent()) {
+			this.server.setParentProcessId(ProcessHandle.current().parent().get().pid());
+		}
```

Comment:
Is it worth guarding this off like we do at https://github.com/eclipse/eclipse.jdt.ls/blob/325726aeb07c1c91d12fe2fda738e82f3bb3e553/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/ParentProcessWatcher.java#L69-L74 to avoid any failures here ?

Reply:
I have never faced any failures at https://github.com/eclipse/eclipse.jdt.ls/blob/325726aeb07c1c91d12fe2fda738e82f3bb3e553/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/ParentProcessWatcher.java#L69-L74
I have added the code to avoid `pid == 0` at https://github.com/eclipse/eclipse.jdt.ls/blob/master/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/ParentProcessWatcher.java#L65